exonum-java-binding/core/src/main/java/com/exonum/binding/core/transaction/TransactionMethod.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.core.transaction;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates that a method is a transaction method. The annotated method should:
+ * <ul>
+ *   <li>be public
+ *   <li>have exactly two parameters of types 'byte[]' and '{@link TransactionContext}'
+ * </ul>
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+// TODO: rename to Transaction after migration?
+public @interface TransactionMethod {
+
+  /**
+   * Returns the transaction type identifier which is unique within the service.
+   */
+  int id();
```

Comment:
As there are no other properties than 'id', it is sensible to use 'value' instead.